velox/exec/tests/RowContainerTest.cpp

```diff
@@ -729,6 +729,37 @@ TEST_F(RowContainerTest, rowSizeWithNormalizedKey) {
   ASSERT_EQ(numRows, 1);
 }
 
+TEST_F(RowContainerTest, estimateRowSize) {
+  auto numRows = 1000;
+
+  // Make a RowContainer with a fixed-length key column and a variable-length
+  // dependent column.
+  auto rowContainer = makeRowContainer({BIGINT()}, {VARCHAR()});
+  EXPECT_FALSE(rowContainer->estimateRowSize().has_value());
+
+  // Store rows to the container.
+  auto input = std::dynamic_pointer_cast<RowVector>(
+      BatchMaker::createBatch(ROW({BIGINT(), VARCHAR()}), numRows, *pool_));
+  SelectivityVector allRows(numRows);
+  DecodedVector decodedKey(*input->childAt(0), allRows);
+  DecodedVector decodedDependent(*input->childAt(1), allRows);
+  for (size_t i = 0; i < numRows; i++) {
+    auto row = rowContainer->newRow();
+    rowContainer->store(decodedKey, i, row, 0);
+    rowContainer->store(decodedDependent, i, row, 1);
+  }
+
+  RowContainerIterator iter;
+  char* result[numRows];
+  auto resultRows = rowContainer->listRows(
```

Comment:
What's the point of listing the rows? Shouldn't we call estimateRowSize instead?

Reply:
I got this idea from TEST_F(rowSizeWithNormalizedKey) and TEST_F(rowSize), and guess it makes sense. As you might know, listRows() returns the first `resultRows` rows that hit the size `rowContainer->estimateRowSize().value() * numRows`. By verifying resultRows == numRows, we see that estimateRowSize() is an accurate estimate.

If there's a better way to verify, would you mind giving an example?

Reply:
@gggrace14 Ge, thank you for explaining. I didn't think about this.